src/base/memory_monitor_sr.cpp

```diff
@@ -20,18 +20,62 @@
  * memory_monitor_sr.cpp - Implementation of memory monitor module
  */
 
+#include <malloc.h>
+#include <cassert>
 #include <cstring>
 #include <algorithm>
-#include <cassert>
 
 #include "memory_monitor_sr.hpp"
 
+#ifndef HAVE_USR_INCLUDE_MALLOC_H
+#define HAVE_USR_INCLUDE_MALLOC_H
+#endif
+
+typedef struct mmon_metainfo
+{
+  uint64_t size;
+  int tag_id;
+  int magic_number;
+} MMON_METAINFO;
+
 namespace cubmem
 {
+  memory_monitor *mmon_Gl = nullptr;
+
   memory_monitor::memory_monitor (const char *server_name)
-    : m_server_name {server_name} {}
+    : m_server_name {server_name}
+  {
+    std::string magic_string ("MMON");
+    memcpy (&magic, magic_string.c_str (), sizeof (int));
```

Comment:
I think `sizeof(int)` is more intuitive than `sizeof(this->magic)`. I want to know why are you think your suggestion is better.

Reply:
Both sizeof(variable) and sizeof(type) are acceptable, but in my opinion, sizeof(variable) is more commonly used for the following reasons:
1. For arrays, you must use sizeof(variable) to get the size of the array. Even if it is not an array, sizeof(variable) should be used to ensure consistent code. (It may be an exception to templates, but historically arrays were introduced first.)
2. If the type of the variable changes when using sizeof(type), finding the location where sizeof() was used is more difficult than when using sizeof(variable).

Reply:
OK. I understand it. Done.